### ISSUE TITLE
Getting Gradle versions from gradle.properties

### ISSUE THREAD

**Issue by MarounMaroun** [dependabot/dependabot-core#1618]
My `build.gradle` has dependencies like:

```java
dependency "org.assertj:assertj-core:${assertJVersion}"
```

The variable `assertJVersion` is resolved from the `gradle.properties` file.

Is there a way to configure the bot to resolve the version from this file?

**Comment by meck93** [comment]
Another option would be to resolve the variable `assertJVersion` via the `ext` section in the `build.gradle`. I'm also not sure if this currently works?

**Comment by matir91** [comment]
From my point of view this is one of the most important issues of this bot, so I think this shouldn't be closed.

**Comment by mercuriete** [comment]
I have dependencies like:

```
    def daggerVersion = 2.20
    implementation "com.google.dagger:dagger:$daggerVersion"
    implementation "com.google.dagger:dagger-android:$daggerVersion"
    implementation "com.google.dagger:dagger-android-support:$daggerVersion"
    kapt "com.google.dagger:dagger-android-processor:$daggerVersion"
    kapt "com.google.dagger:dagger-compiler:$daggerVersion"
    kapt "com.google.dagger:dagger-android-support:$daggerVersion"
    kaptTest "com.google.dagger:dagger-compiler:$daggerVersion"
```

So in this case is mandatory to bump lots of dependencies to the same version and not one by one.

I understand this problem is difficult to resolve, thank in advance!
:+1:

**Comment by ashughes** [comment]
@mercuriete I was very pleasantly surprised that this worked!

**Comment by mercuriete** [comment]
Sorry for the noise if this is offtopic

@ashughes what do you mean about "this worked"?
The snipped that I put was from some tutorial from internet, I am doing something wrong?

The problem is dependabot is not intelligent enough to search def variables and create a PR with my example.

If you mean the workaround that I proposed... I didn't test it yet. I am not sure that dependabot is working with hardcoded versions per each line I dont want to change my master branch to do that test, probably I will end up doing a test on an copy of my repository.

Thanks to the dependabot team for the amazing work.

**Comment by ashughes** [comment]
@mercuriete I wasn't sure whether or not you were suggesting this as working, but I tried it and it _did_ work, so I assumed you were. :stuck_out_tongue: 

I have 3 dependencies that all use the same version, but they each specified the version individually:
```
implementation 'com.fasterxml.jackson.core:jackson-core:2.8.4'
implementation 'com.fasterxml.jackson.core:jackson-annotations:2.8.4'
implementation 'com.fasterxml.jackson.core:jackson-databind:2.8.4'
```

Dependabot created 3 PRs to change each of these individually. After reading your comment I changed the above to the following on my main branch:
```
def jacksonVersion = "2.8.4"
implementation "com.fasterxml.jackson.core:jackson-core:$jacksonVersion"
implementation "com.fasterxml.jackson.core:jackson-annotations:$jacksonVersion"
implementation "com.fasterxml.jackson.core:jackson-databind:$jacksonVersion"
```

Then Dependabot closed all 3 of the previous PRs and opened a new one that just changed `jacksonVersion` to `2.11.1`! And it even included the release notes for all 3 of the dependencies (well, at least as much as it did on the individual PRs).

Basically, doing this appears to be a great workaround for https://github.com/dependabot/dependabot-core/issues/1296.

**Comment by mercuriete** [comment]
@ashughes thanks for the clarification I need to investigate more, because that is what is not working for me.
Maybe I need more coffee xD.

**Comment by mercuriete** [comment]
@ashughes 
Sorry for the noise...
I just changed my snippet from the one is described above... for this one:https://github.com/mercuriete/android-musician-tools/commit/a4de66446f00ba78c95ddc2a18d23662d0d6a5c6


build.gradle on root folder
```
    ext.dagger_version = '2.28.3'
```

build.gradle on app folder
```
    implementation "com.google.dagger:dagger:$dagger_version"
    implementation "com.google.dagger:dagger-android:$dagger_version"
    implementation "com.google.dagger:dagger-android-support:$dagger_version"
    kapt "com.google.dagger:dagger-android-processor:$dagger_version"
    kapt "com.google.dagger:dagger-compiler:$dagger_version"
    kapt "com.google.dagger:dagger-android-support:$dagger_version"
    kaptTest "com.google.dagger:dagger-compiler:$dagger_version"
```

and it started to work.
So sorry for the noise that I introduced here but dependabot works for my use case.

Thanks to all the people involved in this issue.

**Comment by StefanOltmann** [comment]
Any updates on this?

I also define my versions in `gradle.properties` as this feels very right to do so.

**Comment by sschuberth** [comment]
IMO the only sane way to address this is by implementing https://github.com/dependabot/dependabot-core/issues/1164.

**Comment by marcindabrowski** [comment]
I have versions defined as below:

```
project.ext.versions = [
        kotlin                       : '1.5.0',
        kotlin_logging               : '1.12.5',
        wiremock                     : '2.28.0',
]
```

and dependencies :

```
dependencies {
    implementation group: 'io.github.microutils', name: 'kotlin-logging', version: versions.kotlin_logging
    implementation group: 'org.jetbrains.kotlin', name: 'kotlin-reflect', version: versions.kotlin
    implementation group: 'org.jetbrains.kotlin', name: 'kotlin-stdlib', version: versions.kotlin
    implementation group: 'org.jetbrains.kotlin', name: 'kotlin-stdlib-jdk8', version: versions.kotlin
    integration group: 'com.github.tomakehurst', name: 'wiremock-jre8-standalone', version: versions.wiremock
}
```

and it didn't bump kotlin to 1.5.10

**Comment by patkujawa-wf** [comment]
Just saw it work with Kotlin 1.6.0! Thanks again for the tip.

**Comment by mccartney** [comment]
> https://www.sameerkulkarni.de/posts/dependabot-with-gradle/ is the link.

For anyone curious, this gives 404 now.
Web Archive has this post: https://web.archive.org/web/20220125120756/https://www.sameerkulkarni.de/posts/dependabot-with-gradle/

**Comment by SimonScholz** [comment]
> > https://www.sameerkulkarni.de/posts/dependabot-with-gradle/ is the link.
> 
> For anyone curious, this gives 404 now. Web Archive has this post: https://web.archive.org/web/20220125120756/https://www.sameerkulkarni.de/posts/dependabot-with-gradle/

Thanks for sharing this, but IMHO this should be addressed by Dependabot. 
If Gradle can read the versions from the `gradle.properties` file then Dependabot should also be capable of doing so.

**Comment by JakubLedworowski** [comment]
Hi, what is the status of this - is Dependabot's plan to address reading from `gradle.properties`? 
I saw an effort being completed here #6249 which is a nice way of organizing dependencies, but I just wondered if the `gradle.properties` way is also on the roadmap. Thanks.

**Comment by eric-milles** [comment]
It appears from this thread that _dependabot_ can make updates to a property if it is declared within the `dependencies` block of `build.gradle` or in the `ext` block as well.

Is it still an open issue if the property is declared separately in `gradle.properties` or `version.properties`?

Is it beyond hope that it might be able to see past a property name change, like this?

path/to/some.properties:
```
junit.version=5.13.4
```

build.gradle:
```groovy
new Properties().with { props ->
	file('../path/to/some.properties').withInputStream(props.&load)
	for (entry in props) {
		def key = entry.key.replaceAll(/\.\w+/) { it.substring(1).capitalize() }
		this.ext.set(key, entry.value) // junit.version --> junitVersion
	}
}

dependencies {
	testImplementation "org.junit.jupiter:junit-jupiter:$junitVersion"
	testRuntimeOnly    "org.junit.platform:junit-platform-launcher:${junitVersion.replace('5','1')}"
}
```

**Comment by dlehammer** [comment]
Update; I've tried the [ext suggestion](https://github.com/dependabot/dependabot-core/issues/1618#issuecomment-674770068) in https://github.com/dlehammer/gradle-22466-multiple-plugin-test-demo/commit/a5f9c1caa837d27cda0cff19d8656b8eeb0e5599 , unfortunately Dependabot didn't suggest any updates when [triggered](https://github.com/dlehammer/gradle-22466-multiple-plugin-test-demo/network/updates/1256702043) afterwards.

So I've temporarily cluttered the project with a Gradle Version Catalog ( https://github.com/dlehammer/gradle-22466-multiple-plugin-test-demo/commit/b9ab8847792f8fcafe52ee5a23fc19d3bca6d101 ) and that [worked](https://github.com/dlehammer/gradle-22466-multiple-plugin-test-demo/network/updates/1256730857), in-order to utilize Dependabot until native Gradle properties are supported 🤞 🤓
